example02/reducers/reducer.js

```diff
@@ -0,0 +1,34 @@
+const init = {
+    boxes: []
+};
+
+export default function reducer(state = init, action, args) {
+    const random = Math.random;
+    const floor = Math.floor;
+
+    switch (action) {
+        case "ADD_BOX": {
+            const { boxes } = state;
+            return Object.assign({}, state, {
+                boxes: [...boxes, {
+                    top: floor(random() * document.body.clientHeight),
+                    left: floor(random() * document.body.clientWidth),
```

Comment:
How about subtracting the max height and width of a box here so that we don't draw boxes outside of the screen? It currently makes scrollbars appear which in turn changes the layout slightly.

Reply:
I came up with a different solution. Now the app itself is padded, and the box container is `relative` meaning that the boxes should never be able to overflow the container (unless the window is resized).